PlanetaryRobotics/moonranger-rover-messages
# Realign `MOONRANGER_Pose_t` by moving `SeqId` to end of Struct

From PlanetaryRobotics/moonranger-mission-control#157, I have a problem in the cgo code where `SeqId` of `MOONRANGER_Pose_t` is accessible but the rest of the fields are not. This is my proposed fix for this.

## Description

This is required to unblock me in PlanetaryRobotics/moonranger-mission-control#157

For full info, see Slack comment here: https://github.com/PlanetaryRobotics/moonranger-mission-control/issues/157#issuecomment-1234678096

The impact of this is everyone will have to recompile/rebuild and update their rover-messages submodules in their respective repos.

In other words, this is a breaking change.

## How has this been tested?

In `moonranger-mission-control`:
* run it locally (see PlanetaryRobotics/moonranger-mission-control#269)
* See if the error persists or not

I would probably think this should be tested on the rover side...

## Types of changes


## Checklist:



- [x] I have read and followed the `CONTRIBUTING.md` file
- [ ] I have applied the .clang-format file to my changes
- [x] I have commented my code using Doxygen style comments (see Contributing MD for examples), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design documentation 
- [ ] I have updated any relevant parameters in the internal ICD
- [ ] If this is a bug-fix, I have added tests that can catch similar bugs in the future
- [ ] For cfe apps I have reserved an empty address for my msgs_ids and perf_ids from the [tracking list](https://docs.google.com/spreadsheets/d/1yRUhNTRBOiGLswlsWftNaYlwGKDeNQ_fCnDYFrc_7fU/edit#gid=772812381) and checked that it does not clash with other apps
- [ ] For new added message types, there is a timestamp included via either an explicit timestamp variable in the struct or buried in the a tlmHeader.
